test/sequential/sequencerStaking.rewards.test.ts

```diff
@@ -0,0 +1,299 @@
+  const diff = testUser.getWalletDifferences()[0].diff.free;
+  expect(filteredEvent[0]).toEqual(rewardsSessionNumber.toString());
+  expect(diff).bnEqual(sequencerRewards);
+  expect(rewardInfo1).bnEqual(sequencerRewards);
+  expect(rewardInfo2).bnEqual(BN_ZERO);
+});
+
+it("When a sequencer brings an update It will get some points", async () => {
+  await Sudo.batchAsSudoFinalized(Assets.FinalizeTge(), Assets.initIssuance());
+  await createAnUpdate(testUser, chainEth);
+  const rewardsSessionNumber = await getSessionIndex();
+  const pointsValue = await SequencerStaking.points(rewardsSessionNumber);
+  const userAwardedPts = await SequencerStaking.awardedPts(
+    rewardsSessionNumber,
+    testUser.keyRingPair.address,
+  );
+  expect(userAwardedPts).bnEqual(pointsValue);
+});
+
+it("When session ends, tokens will be distributed according the points obtained", async () => {
+  await Sudo.batchAsSudoFinalized(Assets.FinalizeTge(), Assets.initIssuance());
+  const [testUser2, testUser3] = setupUsers();
+  testUser3.addAsset(GASP_ASSET_ID);
+  const minToBeSequencer = await SequencerStaking.minimalStakeAmount();
+  const stakeAndJoinExtrinsicEth =
+    await SequencerStaking.provideSequencerStaking(
+      minToBeSequencer.addn(1000),
+      chainEth,
+    );
+  const stakeAndJoinExtrinsicArb =
+    await SequencerStaking.provideSequencerStaking(
+      minToBeSequencer.addn(1000),
+      chainArb,
+    );
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintNative(testUser2),
+    Assets.mintNative(testUser3),
+    Sudo.sudoAs(testUser2, stakeAndJoinExtrinsicEth),
+    Sudo.sudoAs(testUser3, stakeAndJoinExtrinsicArb),
+  );
+  const txIndexEth = await Rolldown.lastProcessedRequestOnL2(chainEth);
+  const txIndexArb = await Rolldown.lastProcessedRequestOnL2(chainArb);
+  await Rolldown.waitForReadRights(testUser3.keyRingPair.address, 50, chainArb);
+  await waitNewStakingRound();
+  const selectedSequencerEth = (
+    await getApi().query.sequencerStaking.selectedSequencer()
+  ).toHuman().Ethereum;
+  let ethUser1: User;
+  let ethUser2: User;
+  if (selectedSequencerEth!.toString() === testUser.keyRingPair.address) {
+    ethUser1 = testUser;
+    ethUser2 = testUser2;
+  } else {
+    ethUser1 = testUser2;
+    ethUser2 = testUser;
+  }
+  ethUser1.addAsset(GASP_ASSET_ID);
+  ethUser2.addAsset(GASP_ASSET_ID);
+  const updateEvents = await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAsWithAddressString(
+      ethUser1.keyRingPair.address,
+      await getUpdate(txIndexEth, ethUser1.keyRingPair.address, chainEth),
+    ),
+    Sudo.sudoAsWithAddressString(
+      testUser3.keyRingPair.address,
+      await getUpdate(txIndexArb, testUser3.keyRingPair.address, chainArb),
+    ),
+  );
+  const waitingBlockNumber =
+    (await Rolldown.getRequestIdFromEvents(updateEvents)) + 1;
+  await waitBlockNumber(waitingBlockNumber.toString(), 50);
+  const rewardsSessionNumber = await getSessionIndex();
+  await Sudo.batchAsSudoFinalized(
```

Comment:
We've discussed a scenario in which we perform an operation in the Ethereum network with different selected sequencers in the Arbitrum network. I assume that since the time of the dispute is equal to half the time of the session, the first function is performed at the very beginning and the second after the period of the dispute, all this happens in one session. We also check this by counting the rewards at the end of the test.

Reply:
ok! hopefully this goes right! :crossed_fingers: